app/content/models/registration.py

```diff
@@ -208,10 +205,25 @@ def should_swap_with_non_prioritized_user(self):
     def is_prioritized(self):
         if self.user.number_of_strikes >= 3:
             return False
+
         user_class, user_study = EnumUtils.get_user_enums(**self.user.__dict__)
-        return self.event.registration_priorities.filter(
-            user_class=user_class, user_study=user_study
-        ).exists()
+        user_query = Q(user_class=user_class, user_study=user_study)
+
+        is_in_priority_pool = False
+        user_groups = set(self.user.group_members.values_list("slug", flat=True))
+        pools = self.event.priority_pools.prefetch_related("groups").all()
+
+        for pool in pools:
+            pool_groups = set(pool.groups.values_list("slug", flat=True))
+            is_in_priority_pool = pool_groups.issubset(user_groups) and len(pool_groups)
+
+            if is_in_priority_pool:
+                break
+
+        return (
+            self.event.registration_priorities.filter(user_query).exists()
+            or is_in_priority_pool
+        )
```

Comment:
Dette er kanskje enda mer clean?
```suggestion
        user_groups = set(self.user.group_members.values_list("slug", flat=True))
        pools = self.event.priority_pools.prefetch_related("groups").all()

        for pool in pools:
            pool_groups = set(pool.groups.values_list("slug", flat=True))
            is_in_priority_pool = pool_groups.issubset(user_groups) and len(pool_groups)

            if is_in_priority_pool:
                return True

        return self.event.registration_priorities.filter(user_query).exists()
```